src/ips.rs

```diff
@@ -0,0 +1,525 @@
+                peer_ratings.sort_by(|a, b| b.2.partial_cmp(&a.2).unwrap());
+
+                // Add peers with highest rating as candidates
+                for peer in peer_ratings.iter() {
+                    // Check if peer is already in peerlist - if so go to next one
+                    if peer_list_entry.list.contains(&peer.0) {
+                        continue;
+                    }
+
+                    candidates.push(*peer);
+                    candidates_to_search -= 1;
+                    if candidates_to_search == 0 {
+                        break;
+                    }
+                }
+
+                // Here we have 2*peers_to_add_count candidates to add sorted by ranking.
+                // We need to choose best ones from them - let's choose those with lowest
+                // betweenness factor - just to avoid creating "hot" nodes that have very high
+                // importance to the network which can be risky if such node goes down.
+                candidates.sort_by(|a, b| {
+                    state.nodes[a.1]
+                        .betweenness
+                        .partial_cmp(&state.nodes[b.1].betweenness)
+                        .unwrap()
+                });
+
+                for peer in candidates.iter().take(peers_to_add_count as usize) {
+                    peer_list_entry.list.push(peer.0);
+                }
+            }
+
+            // Do not compute factors one more time after every single peerlist addition. At least
+            // for now, when computing factors is very expensive (especially betweenness and closeness).
+            // Re-calculating it after each node for whole graph would take too long.
+
+            peer_list.push(peer_list_entry);
+        }
+        peer_list
+    }
+
+    // Helper functions
+
+    /// Update nodes rating based on location
+    fn update_rating_by_location(
+        &self,
+        selected_node: &Node,
+        nodes: &[Node],
+        ratings: &mut [(IpAddr, usize, f64)],
+    ) {
+        let selected_location = &selected_node
+            .geolocation
+            .as_ref()
+            .map(|geo_info| {
+                Location::new(
+                    geo_info.latitude.unwrap_or_default(), // TODO(asmie): refactor when zg-core will be updated with Location
+                    geo_info.longitude.unwrap_or_default(),
+                )
+            })
+            .unwrap_or(Location::new(0.0, 0.0));
+
+        for (node_idx, node) in nodes.iter().enumerate() {
+            let mut rating = 0.0;
+            if let Some(geo_info) = &node.geolocation {
+                let location = Location::new(
+                    geo_info.latitude.unwrap_or_default(),
+                    geo_info.longitude.unwrap_or_default(),
+                );
+                let distance = selected_location.distance_to(&location).unwrap().meters();
+                let pref_distance = self.config.geolocation_minmax_distance_km as f64 * 1000.0;
```

Comment:
As the struct field was changed, perhaps the var should mirror this?

Reply:
Changed.